### PR TITLE
[Keymap] Add ariccb keymap to Planck/Rev6

### PR DESCRIPTION
**Pull request by ariccb**
## Description
Added my own custom keymap for the planck rev6

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/ariccb/keymap.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/planck/keymaps/ariccb/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/planck/keymaps/ariccb/keymap.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+|HYPER,Tab|  a  |  r  |  s  |  t  |  g  |  h  |  n  |  e  |  i  |  o  |  ' |
+ * |-----------------------------------------------------------------------|
+ * |shift |  z  |  x  |  c  |  d  |  v  |  k  |  m  |  ,  |  .  |  /  |  \ |
+ * |-----------------------------------------------------------------------|
+ * | Ctl  | Win | Alt |Enter|LOWER|   Space   |RAISE|Menu |Play | Lft |Rght|
+ * `-----------------------------------------------------------------------'
+ */
+[_COLEMAK] = LAYOUT_planck_grid( /* COLEMAK */
+    LT(_FN, KC_ESC),    KC_Q,    KC_W,    KC_F,   KC_P,  KC_B,   KC_J,   KC_L,  KC_U,    KC_Y,    KC_SCLN, KC_BSPC,
+    LT(KC_HYPR, KC_TAB),KC_A,    KC_R,    KC_S,   KC_T,  KC_G,   KC_M,   KC_N,  KC_E,    KC_I,    KC_O,    KC_QUOT,
```
Comment (drashna) [inline keyboards/planck/keymaps/ariccb/keymap.c:79]: LT require a layer, not a keycode.

File: keyboards/planck/keymaps/ariccb/rules.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+MIDI_ENABLE = yes           # MIDI controls
+AUDIO_ENABLE = yes          # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
+TAP_DANCE_ENABLE = no 		# Enable Tap Dance Features
+LTO_ENABLE = no 			# Link Time Optimization
```
Comment (drashna) [inline keyboards/planck/keymaps/ariccb/rules.mk:18]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
MIDI_ENABLE = yes           # MIDI controls
AUDIO_ENABLE = yes          # Audio output
RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
TAP_DANCE_ENABLE = no 		# Enable Tap Dance Features
LTO_ENABLE = no 			# Link Time Optimization
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/ariccb/keymap.c (hunk 1 above)
Comment (ariccb) [reply keyboards/planck/keymaps/ariccb/keymap.c:1]: Yeah absolutely, sorry I missed this!

File: keyboards/planck/keymaps/ariccb/keymap.c (hunk 2 above)
Comment (ariccb) [reply keyboards/planck/keymaps/ariccb/keymap.c:79]: oops, was intending on a Mod-Tap there, not a layer switch.

File: keyboards/planck/keymaps/ariccb/rules.mk (hunk 3 above)
Comment (ariccb) [reply keyboards/planck/keymaps/ariccb/rules.mk:18]: Will do.

**Comment by drashna** [comment]
https://docs.qmk.fm/#/feature_st7565

yes, you can

**Comment by ariccb** [comment]
Hi @drashna, is this able to be merged now?

**Comment by zvecr** [comment]
> is this able to be merged now?

Not until the PR history is corrected. Changes to things like the submodules are instantly going to stop this from being eligible.

**Comment by ariccb** [comment]
> > is this able to be merged now?
> 
> Not until the PR history is corrected. Changes to things like the submodules are instantly going to stop this from being eligible.

What issues are there in the Pull Request history? Not sure what changes were made in the submodules. I apologize for my ignorance, this is my first pull request! I'm still learning the process. :)

**Comment by drashna** [comment]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Comment by ariccb** [comment]
Oh jeez. And I do this on the master branch of my forked repo right? Thanks for the help getting this resolved.



Aric Crosson Bouwers



-------- Original message --------
From: Drashna Jaelre ***@***.***>
Date: 2022-01-21 8:28 p.m. (GMT-07:00)
To: qmk/qmk_firmware ***@***.***>
Cc: Aric Crosson Bouwers ***@***.***>, Author ***@***.***>
Subject: Re: [qmk/qmk_firmware] [Keymap] Add ariccb keymap to Planck/Rev6 (PR #15293)


It looks like you have changes to the submodules here. These need to be reverted before the PR can be approved.

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:

git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234

—
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fqmk%2Fqmk_firmware%2Fpull%2F15293%23issuecomment-1019031615&data=04%7C01%7C%7C1e4bfaec493e4c64723608d9dd5734d2%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637784188894503327%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=cqCI5gX6msbSFylPpUSGCPdY%2FkErGmzx2qV4eKnEWto%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALVQHP6OTR4WYVE5ANKMCTTUXIP4NANCNFSM5IXGNHDA&data=04%7C01%7C%7C1e4bfaec493e4c64723608d9dd5734d2%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637784188894503327%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=kqKCISAK0HdKg5MYgTNZoiJjIl4MXa0F8BmVxj%2B%2Bi5g%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7C1e4bfaec493e4c64723608d9dd5734d2%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637784188894503327%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=mVHKwK2P%2Bz9ORUvo71vbVd7QODzadvoNBF36A6ei%2Bus%3D&reserved=0> or Android<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7C1e4bfaec493e4c64723608d9dd5734d2%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637784188894503327%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=ZpZtuRlXweLF89K%2FPMV%2Fd33UxXAAhBUqjPsPrt%2B7v%2Bc%3D&reserved=0>.

**Comment by drashna** [comment]
Oh, you're ... using your master branch for the PR.  We don't recommend doing that for a number of reasons.

So instead of `git checkout master` you want `git checkout upstream/master` instead.  Ignore the message about a detached head. It's not important here, basically.  The rest should be the same.

**Comment by ariccb** [comment]
I accidentally started the pull request on that instead of my Development branch, is it possible to simply cancel this PR and start a new one?

**Comment by ariccb** [comment]
> Oh, you're ... using your master branch for the PR. We don't recommend doing that for a number of reasons.
> 
> So instead of `git checkout master` you want `git checkout upstream/master` instead. Ignore the message about a detached head. It's not important here, basically. The rest should be the same.

when I try to git checkout upstream/master, I get an error saying it did not match any files known to git 
image

**Comment by ariccb** [comment]
> ```shell
> # for each of the problematic repo's:
> cd lib/@@name@@
> git checkout abcdef1234
> ```

For the part you mention # for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234, 

I'm not sure what you mean by cd lib/@@name@@, is @@name@@ supposed to be the name of the folder that's having an issue? or am I supposed to have a folder there called ariccb? 
also, not sure what "the first abcdef1234" means, do you mean the files that have the issue? When I ran it the first time I got this: 
`aricb@BouwersLenovo MINGW64 ~/qmk_firmware/keyboards/planck/keymaps/ariccb (master)
$ git submodule foreach 'git log -nl --oneline'
Entering '../../../../lib/chibios'
Entering '../../../../lib/chibios-contrib'
Entering '../../../../lib/googletest'
Entering '../../../../lib/lufa'
Entering '../../../../lib/printf'
fatal: No url found for submodule path '../../../../lib/ugfx' in .gitmodules`

so is 'ugfx' the 'abcdef1234' you're talking about or is '.gitmodules' the 'abcdef1234'?
I tried both and got nowhere... gotta be honest I have only really used GitKraken for my git so far, haven't worked in the CLI yet
image

**Comment by drashna** [comment]
Remove ugfx.

**Comment by ariccb** [comment]
> Remove ugfx.

managed to get this. What is the next step to fix the error at lib/ugfx? 
image

**Pull request closed without merging**